### ISSUE TITLE
Sculk snapper breeding crash

### ISSUE THREAD

**Issue by NewJumper** [KyaniteMods/DeeperAndDarker#102]
**Description**:
Using an enchanted netherite chestplate on an already tame snapper causes the came to crash.

**To Reproduce**:
1. Tame a snapper using an enchanted chestplate.
2. In survival or creative mode, keep right-clicking the snapper with another enchanted chestplate.
3. crash!

forge-v1.1.3-413

**Comment by NewJumper** [comment]
have not testing on fabric

**Comment by LinuxSquare** [comment]
> i have not tested on fabric

I can confirm, that it happens also on fabric in version 1.1.6.

Here is a short snippet from my crashlog.
```
---- Minecraft Crash Report ----
// Everything's going to plan. No, really, that was supposed to happen.

Time: 2022-12-12 21:07:10
Description: Ticking entity

java.lang.ClassCastException: class net.minecraft.class_5321 cannot be cast to class java.util.List (net.minecraft.class_5321 is in unnamed module of loader net.fabricmc.loader.impl.launch.knot.KnotClassLoader @327471b5; java.util.List is in module java.base of loader 'bootstrap')
	at com.kyanite.deeperdarker.registry.entities.custom.SculkSnapperEntity.method_5773(SculkSnapperEntity.java:130)
	at net.minecraft.class_638.method_18646(class_638.java:267)
	at net.minecraft.class_1937.method_18472(class_1937.java:466)
	at net.minecraft.class_638.method_32124(class_638.java:250)
	at net.minecraft.class_5574.method_31791(class_5574.java:54)
	at net.minecraft.class_638.method_18116(class_638.java:246)
	at net.minecraft.class_310.method_1574(class_310.java:1851)
	at net.minecraft.class_310.method_1523(class_310.java:1135)
	at net.minecraft.class_310.method_1514(class_310.java:768)
	at net.minecraft.client.main.Main.method_44604(Main.java:244)
	at net.minecraft.client.main.Main.main(Main.java:51)
	at net.fabricmc.loader.impl.game.minecraft.MinecraftGameProvider.launch(MinecraftGameProvider.java:461)
	at net.fabricmc.loader.impl.launch.knot.Knot.launch(Knot.java:74)
	at net.fabricmc.loader.impl.launch.knot.KnotClient.main(KnotClient.java:23)
```
In the appendix, you'll find both crash-logs.

[crash-2022-12-12_21.09.41-server.txt](https://github.com/KyaniteMods/DeeperAndDarker/files/10211662/crash-2022-12-12_21.09.41-server.txt)
[crash-2022-12-12_21.07.10-client.txt](https://github.com/KyaniteMods/DeeperAndDarker/files/10211663/crash-2022-12-12_21.07.10-client.txt)

**Comment by nitrodynamite18** [comment]
Finally fixed by commit #1006.
